### PR TITLE
Added the algorithms to extract relationhsip

### PR DESCRIPTION
**Pull request by ansjin**
Added these two algorithms to take either file or an array of sentences and return the resulted values in a proper json format as shown in readme
1. Ollie
2. OpenIE

This algorithms will run inside separate docker container. 
Node.js interface is added to interact with these algorithms.

Also updated the readme and added the config file.

### CODE REVIEW

**Review by RBirkeland — APPROVED** [review]
Nice!

File: app/algorithms/ollie/index.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+var port     = process.env.PORT || config.server.port; // set our port
+
+
+//Ollie Algorithm
+//=============================================================================================================================
+
+var ollie = express.Router();
+// middleware to use for all requests
+ollie.use(function(req, res, next) {
+  // do logging
```
Comment (kordianbruck) [inline app/algorithms/ollie/index.js:29]: Not a useful comment

File: app/algorithms/ollie/index.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      var arr = req.body.array;
+      if(!arr)
+      {
+        inputFile = config.ollieAlgo.defaultFilePath;
+      }
+      else {
+        inputFile = config.ollieAlgo.defaultFileSavePath;
+
+        var file = fs.createWriteStream(inputFile);
+        file.on('error', function (err) { /* error handling */
```
Comment (kordianbruck) [inline app/algorithms/ollie/index.js:51]: Not a useful comment.

File: app/algorithms/ollie/index.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    var relations=
+      {
+        sentence:"",
+        instances:[
+          {
+            quality:"",
+            term1:"",
+            term2:"",
+            term3:""
+            /*relation:"",*/
```
Comment (kordianbruck) [inline app/algorithms/ollie/index.js:73]: Why is this commented? If not needed then remove it?

File: app/algorithms/ollie/index.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+    var instancesArr = [];
+    var exMessage = [];
+    exec(command,function (error, stdout, stderr)
+    {
+      var lines = stdout.toString().replace(/([.?!])\s*(?=[A-Z])/g, "\n").split("\n");
+      for(var line in lines)
+      {
+        lines[line] = lines[line].replace("\r", "");
+        //console.log(line)
```
Comment (kordianbruck) [inline app/algorithms/ollie/index.js:86]: This really should not be merged with those lines

File: app/algorithms/ollie/index.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (kordianbruck) [inline app/algorithms/ollie/index.js:141]: Are those empty lines at the end of the file on purpose?

File: app/algorithms/ollie/package.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  "license": "ISC",
+  "dependencies": {
+    "body-parser": "^1.17.1",
+    "child_process": "^1.0.2",
+    "clone": "^2.1.1",
+    "express": "^4.15.2",
+    "fs": "0.0.1-security",
+    "morgan": "^1.8.1",
+    "request": "^2.81.0",
+    "tail": "^1.2.1"
```
Comment (kordianbruck) [inline app/algorithms/ollie/package.json:20]: Please lock down the versions or use yarn https://yarnpkg.com

File: app/algorithms/openie/config/index.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+// config.js
```
Comment (kordianbruck) [inline app/algorithms/openie/config/index.js:1]: Make this a json file please. That way you don't need to export it.

Comment (kordianbruck) [reply app/algorithms/openie/config/index.js:1]: Plus no one will dream of adding logic here

File: app/algorithms/openie/index.js [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+           else {
+             relations.instances = instancesArr;
+             var temp = clone(relations);
+             exMessage.push(temp);
+             //console.log(relations);
+             relations.sentence = sentence;
+             instancesArr = [];
+           }
+         }
+         //else
```
Comment (kordianbruck) [inline app/algorithms/openie/index.js:106]: Really? What is this?

**Review by kordianbruck — COMMENTED** [review]
Ouch - how did get merged without any feedback? Please actually use the review system instead of just approving the changes!

**Pull request merged**